server/embed/config_test.go

```diff
@@ -92,136 +92,99 @@ func TestConfigFileOtherFields(t *testing.T) {
-			experimentalInitialCorruptCheck: "true",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.StopGRPCServiceOnDefrag: false,
-				features.DistributedTracing:      false,
-				features.InitialCorruptCheck:     true,
-			},
-		},
-		{
-			name:                            "can set feature gate experimentalInitialCorruptCheck to false from experimental flag",
-			experimentalInitialCorruptCheck: "false",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.StopGRPCServiceOnDefrag: false,
-				features.DistributedTracing:      false,
-				features.InitialCorruptCheck:     false,
-			},
-		},
-		{
-			name:                   "can set feature gate StopGRPCServiceOnDefrag to true from feature gate flag",
-			serverFeatureGatesJSON: "StopGRPCServiceOnDefrag=true",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.StopGRPCServiceOnDefrag: true,
-				features.DistributedTracing:      false,
-				features.InitialCorruptCheck:     false,
-			},
+			name:                   "can set feature gate to false from feature gate flag",
+			serverFeatureGatesJSON: fgname + "=false",
+			expectedFeatures:       defaultFeatureGatesWithFeaturegate(fg, false),
 		},
+	}
+}
+
+type featureTestCase struct {
+	name                   string
+	serverFeatureGatesJSON string
+	expectErr              bool
+	expectedFeatures       map[featuregate.Feature]bool
+	experimentalKey        string
+	experimentalValue      bool
+}
+
+func TestConfigFileFeatureGates(t *testing.T) {
+	testCases := []featureTestCase{
 		{
-			name:                   "can set feature gate InitialCorruptCheck to true from feature gate flag",
-			serverFeatureGatesJSON: "InitialCorruptCheck=true",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.StopGRPCServiceOnDefrag: false,
-				features.DistributedTracing:      false,
-				features.InitialCorruptCheck:     true,
-			},
+			name:             "default",
+			expectedFeatures: defaultFeatureGates(),
 		},
 		{
-			name:                   "can set feature gate to false from feature gate flag",
-			serverFeatureGatesJSON: "StopGRPCServiceOnDefrag=false",
-			expectedFeatures: map[featuregate.Feature]bool{
-				features.StopGRPCServiceOnDefrag: false,
-				features.DistributedTracing:      false,
-				features.InitialCorruptCheck:     false,
-			},
+			name:                   "ok to set different experimental flag and feature gate flag",
+			serverFeatureGatesJSON: "DistributedTracing=true",
+			experimentalKey:        "experimental-stop-grpc-service-on-defrag",
+			experimentalValue:      true,
+			expectedFeatures:       defaultFeatureGatesWithFeaturegates([]featuregate.Feature{features.StopGRPCServiceOnDefrag, features.DistributedTracing}, true),
 		},
 	}
+
+	testCases = append(testCases, testCasesForFeaturegate("experimental-stop-grpc-service-on-defrag", "StopGRPCServiceOnDefrag", features.StopGRPCServiceOnDefrag)...)
```

Comment:
Any reason for appends here?

Reply:
Add a comment for it, how aboud now.

```
// add a new experimental flag here when migrating a new experimental flag to feature.
```

Reply:
I meant that why some cases are defined as at list creation and why some are appends? Append is more concise, could we use it for declaring the whole `testCases` slice?